src/table/components/virtualized-table/hooks/use-infinite-scroll-data.ts

```diff
@@ -47,8 +48,7 @@ const useInfiniteScrollData = (
   pageInfo: PageInfo
 ): UseInfiniteScrollData => {
   const [rowsInPage, setRowsInPage] = useState<Row[]>([]);
-
-  useEffect(() => setRowsInPage([]), [layout, pageInfo]);
+  useOnPropsChange(() => setRowsInPage([]), [layout, pageInfo]);
```

Comment:
By not using useEffect here we get 1 less render cycle. Come out of me trying to fix the original issue but ended up being valuable.